constantine/eth_verkle_ipa/barycentric_form.nim

```diff
@@ -29,7 +29,7 @@ func newPrecomputedWeights* [PrecomputedWeights] (res: var PrecomputedWeights) =
 
   var midpoint = VerkleDomain
   for i in 0 ..< midpoint:
-    var weights {.noInit.}: Fr[Banderwagon]
+    var weights: Fr[Banderwagon]
```

Comment:
Is this related to the random garbage you mentioned on Discord https://discord.com/channels/613988663034118151/1138059404831379508/1210186486662565888?

computeBarycentricWeights below is always initialized so it's strange that it fixes uninitialized memory: https://github.com/mratsim/constantine/blob/c7979b003372b329dd450ff152bb5945cafb0db5/constantine/eth_verkle_ipa/barycentric_form.nim#L62-L81

Reply:
Nope this wasn't the place, but yes the final value of `weights` shouldn't change any of the values as we're iterating through the `VerkleDomain`, ok i can revert it back.